scripts/auth_init.py

```diff
@@ -0,0 +1,177 @@
+                return response_json["secretText"]
+
+            raise Exception(response_json)
+
+
+async def create_or_update_application_with_secret(
+    auth_headers: Dict[str, str], app_id_env_var: str, app_secret_env_var: str, app_payload: Dict[str, Any]
+) -> Tuple[str, str, bool]:
+    app_id = os.getenv(app_id_env_var, "no-id")
+    created_app = False
+    object_id = None
+    if app_id != "no-id":
+        print(f"Checking if application {app_id} exists")
+        object_id = await get_application(auth_headers, app_id)
+
+    if object_id:
+        print("Application already exists, not creating new one")
+        await update_application(auth_headers, object_id, app_payload)
+    else:
+        print("Creating application registration")
+        object_id, app_id = await create_application(auth_headers, app_payload)
+        update_azd_env(app_id_env_var, app_id)
+        created_app = True
+
+    if object_id and os.getenv(app_secret_env_var, "no-secret") == "no-secret":
+        print(f"Adding client secret to {app_id}")
+        client_secret = await add_client_secret(auth_headers, object_id)
+        update_azd_env(app_secret_env_var, client_secret)
+
+    return (object_id, app_id, created_app)
+
+
+def update_azd_env(name, val):
+    subprocess.run(f"azd env set {name} {val}", shell=True)
+
+
+def create_server_app_initial_payload():
+    return {
+        "displayName": "Azure Search OpenAI Demo Server App",
+        "signInAudience": "AzureADandPersonalMicrosoftAccount",
+    }
+
+
+def create_server_app_permission_setup_payload(server_app_id: str):
+    return {
+        "api": {
+            "knownClientApplications": [],
+            "oauth2PermissionScopes": [
+                {
+                    "id": "7b207263-0c4a-4127-a6fe-38ea8c8cd1a7",
+                    "adminConsentDisplayName": "Access Azure Search OpenAI Demo API",
+                    "adminConsentDescription": "Allows the app to access Azure Search OpenAI Demo API as the signed-in user.",
+                    "userConsentDisplayName": "Access Azure Search OpenAI Demo API",
+                    "userConsentDescription": "Allow the app to access Azure Search OpenAI Demo API on your behalf",
+                    "isEnabled": True,
+                    "value": "access_as_user",
+                    "type": "User",
+                }
+            ],
+        },
+        "requiredResourceAccess": [
+            # Graph User.Read
+            {
+                "resourceAppId": "00000003-0000-0000-c000-000000000000",
+                "resourceAccess": [{"id": "e1fe6dd8-ba31-4d61-89e7-88639da4683d", "type": "Scope"}],
+            }
+        ],
+        "identifierUris": [f"api://{server_app_id}"],
+    }
+
+
+def create_client_app_payload(server_app_id: str, server_app_permission_setup_payload: Dict[str, Any]):
+    return {
+        "displayName": "Azure Search OpenAI Demo Client App",
+        "signInAudience": "AzureADandPersonalMicrosoftAccount",
```

Comment:
This will allow anyone in any organization with an AD tenant or anyone with a personal MSA account to try and  log in to the application.  Is this the intent?  I'd suggest setting this by default to `AzureADMyOrg` so only members of the current organization can connect.

Reply:
@pamelafox  in your branch you used this constant, are you OK switching it to AzureADMyOrg?

Reply:
Yes, that's probably what most expect, they can broaden it if desired.